### PR TITLE
Replace RequireJS with Almond during build [fixes #296]

### LINKED ISSUE #296: Generator creates usemin block for require which is no longer supported.
Getting all set to build and ran "grunt", got this error 'Fatal error: require.js blocks are no more supported.'.  Found this ... https://github.com/yeoman/grunt-usemin/issues/112.  Are there plans to change the scaffold in regards to this incompatibility?

### PR DESCRIPTION
**Pull request by marian-r**
Replacement for #299 & #300.

Generated output in some configurations can be seen here: https://github.com/marian-r/generator-backbone-examples.

### CODE REVIEW

File: generators/app/templates/index.html [hunk 1]
```diff
@@ -107,5 +107,8 @@ <h3 class="text-muted"><%= appName %></h3>
       r.parentNode.insertBefore(e,r)}(window,document,'script','ga'));
       ga('create','UA-XXXXX-X');ga('send','pageview');
     </script>
+<% if (includeRequireJS) { -%>
+    <script data-main="scripts/main" src="bower_components/requirejs/require.js"></script>
+<% } -%>
```
Comment (ruyadorno) [inline generators/app/templates/index.html]: simple and clever solution to bypass the usemin block generated by `htmlWiring.appendScript` :+1: nice job!

**Comment by ruyadorno** [comment]
:+1: LGTM

thank you very much @marian-r for the awesome work putting this all together

happy to see the requirejs build finally working again :confetti_ball:

File: generators/app/templates/index.html (hunk 1 above)
Comment (marian-r) [inline generators/app/templates/index.html]: Thanks :) I saw that append was producing incorrect indentation, so I fixed it with this simple solution.

**Comment by marian-r** [comment]
Thanks for your patience :+1:

**Pull request merged**